src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -0,0 +1,122 @@
+// <auto-generated/>
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, Program.MyClass obj) => BindCore(configuration, ref obj);
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        if (configuration["MyString"] is string stringValue0)
+        {
+            obj.MyString = stringValue0;
+        }
+
+        if (configuration["MyInt"] is string stringValue1)
+        {
+            obj.MyInt = int.Parse(stringValue1);
+        }
+
+        System.Collections.Generic.List<int> temp2 = obj.MyList;
+        temp2 ??= new System.Collections.Generic.List<int>();
+        BindCore(configuration.GetSection("MyList"), ref temp2);
+        obj.MyList = temp2;
+
+        System.Collections.Generic.Dictionary<string, string> temp3 = obj.MyDictionary;
+        temp3 ??= new System.Collections.Generic.Dictionary<string, string>();
+        BindCore(configuration.GetSection("MyDictionary"), ref temp3);
+        obj.MyDictionary = temp3;
+
+        System.Collections.Generic.Dictionary<string, Program.MyClass2> temp4 = obj.MyComplexDictionary;
+        temp4 ??= new System.Collections.Generic.Dictionary<string, Program.MyClass2>();
+        BindCore(configuration.GetSection("MyComplexDictionary"), ref temp4);
+        obj.MyComplexDictionary = temp4;
+
+    }
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref System.Collections.Generic.List<int> obj)
```

Comment:
>ref [](http://example.com/codeflow?start=106&length=3)

why `ref` is needed here?

Reply:
It is needed when binding to structs and I found that in some cases, though I don't remember the specifics, it is also needed for some reference types. So we just use `ref` in all cases for both functionality and consistency.